File: YukariLauncher/src/main/java/net/kdt/pojavlaunch/customcontrols/keyboard/LwjglCharSender.java
package net.kdt.pojavlaunch.customcontrols.keyboard;

import static org.lwjgl.glfw.CallbackBridge.sendKeyPress;

import net.kdt.pojavlaunch.LwjglGlfwKeycode;

import org.lwjgl.glfw.CallbackBridge;

/** Sends keys via the CallBackBridge */
public class LwjglCharSender implements CharacterSenderStrategy {
/**
 * 「send Backspace」処理を実行します。
 * このメソッドは特定の機能を提供するために実装されています。
 */
    @Override
    public void sendBackspace() {
        CallbackBridge.sendKeycode(LwjglGlfwKeycode.GLFW_KEY_BACKSPACE, '\u0008', 0, 0, true);
        CallbackBridge.sendKeycode(LwjglGlfwKeycode.GLFW_KEY_BACKSPACE, '\u0008', 0, 0, false);
    }
/**
 * 「send Enter」処理を実行します。
 * このメソッドは特定の機能を提供するために実装されています。
 */
    @Override
    public void sendEnter() {
        sendKeyPress(LwjglGlfwKeycode.GLFW_KEY_ENTER);
    }
/**
 * 「send Char」処理を実行します。
 * このメソッドは特定の機能を提供するために実装されています。
 */
    @Override
    public void sendChar(char character) {
        CallbackBridge.sendChar(character, 0);
    }
}

File: YukariLauncher/src/main/java/com/arata/yukarilauncher/ui/subassembly/customcontrols/keyboard/LwjglCharSender.kt
package com.arata.yukarilauncher.ui.subassembly.customcontrols.keyboard

import com.arata.yukarilauncher.utils.LwjglGlfwKeycode
import org.lwjgl.glfw.CallbackBridge

/** LWJGL経由で文字をMinecraftエンジンに送信する実装。 */
class LwjglCharSender : CharacterSenderStrategy {
    override fun sendBackspace() {
        CallbackBridge.sendKeycode(LwjglGlfwKeycode.GLFW_KEY_BACKSPACE.toInt(), '\u0008', 0, 0, true)
        CallbackBridge.sendKeycode(LwjglGlfwKeycode.GLFW_KEY_BACKSPACE.toInt(), '\u0008', 0, 0, false)
    }

    override fun sendEnter() {
        CallbackBridge.sendKeyPress(LwjglGlfwKeycode.GLFW_KEY_ENTER.toInt())
    }

    override fun sendChar(character: Char) {
        CallbackBridge.sendChar(character, 0)
    }
}
